ospd/ospd.py

```diff
@@ -584,23 +585,40 @@ def remove_scan_hosts_from_target_progress(
             scan_id, exc_hosts_list
         )
 
-    def set_scan_host_finished(self, scan_id: str, host: str) -> None:
+    def set_scan_host_finished(
+        self,
+        scan_id: str,
+        host: str = None,
+        finished_host_batch: List[str] = None,
+    ) -> None:
```

Comment:
```suggestion
    def set_scan_host_finished(
        self,
        scan_id: str,
        finished_hosts: Optional[Union[List[str], str]] = None,
    ) -> None:
```

Reply:
It is not a good practice to have dependencies between arguments like host and finished_host_batch. A passed argument should not be ignored and overridden because some other argument has been passed.